cloudinit/util.py

```diff
@@ -68,17 +68,31 @@
                    ['running-in-container'],
                    ['lxc-is-container'])
 
-PROC_CMDLINE = None
+_CACHED_RESPONSES = {}
```

Comment:
What's the thinking behind using a global here, instead of making memoize a class with an instance for each function being memoized (which can store the state internally)?